geo: prepare for passing ARM tests

otan opened the pull request:
See individual commits for details

----

cc @rickystewart

otan committed:
geomfn: fix small bugs in ST_LineSubString

Release note (bug fix): Fix a bug where a substring of a linestring with
the same points would return EMPTY instead of a POINT with the repeated
point.

Release note (bug fix): Fix a bug where it is possible for a linestring
returned from ST_LineSubString to have repeated points.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/66621)
<!-- Reviewable:end -->

otan committed:
geo: use epsilon in more places

ARM seems to round differently to results we typically have. To support
this, we should use epsilon comparisons more often. We move some of this
functionality to a `geotest` package.

We also have to write our own `InEpsilon`, as this require.InEpsilon will
fail if 0 is used as an expected value.

This gets everything except for the geoindex tests that failed
[here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_RickyTestForArm/3095227).

Release note: None

otan committed:
geoindex: change keysToString to print bbox to 6 decimal places

This allows a small epsilon for test results.

Release note: None

otan commented:
pausing development on this for now! if ARM turns out to be a requirement i'll revive this.

prashantkhoje commented:
I can see approximation allowed for s390x at https://github.com/cockroachdb/cockroach/commit/688583e26aa3284cacce8d5aa7feaec881f4ed6a
I wonder if we can do the same for other platforms and wider set of tests.
